src/circuit/gadget/ecc/chip/witness_scalar_fixed/short.rs

```diff
@@ -0,0 +1,95 @@
+use super::super::{CellValue, EccConfig, EccScalarFixedShort, Var};
+use crate::constants::{L_VALUE, NUM_WINDOWS_SHORT};
+use ff::PrimeFieldBits;
+use halo2::{
+    arithmetic::{CurveAffine, Field, FieldExt},
+    circuit::Region,
+    plonk::{ConstraintSystem, Error, Expression, Selector},
+    poly::Rotation,
+};
+
+pub struct Config<C: CurveAffine> {
+    q_scalar_fixed_short: Selector,
+    super_config: super::Config<C>,
+}
+
+impl<C: CurveAffine> From<&EccConfig<C>> for Config<C> {
+    fn from(ecc_config: &EccConfig<C>) -> Self {
+        Self {
+            q_scalar_fixed_short: ecc_config.q_scalar_fixed_short,
+            super_config: ecc_config.into(),
+        }
+    }
+}
+
+impl<C: CurveAffine> Config<C> {
+    pub fn create_gate(&self, meta: &mut ConstraintSystem<C::Base>) {
+        // Check that sign is either 1 or -1.
+        meta.create_gate("check sign", |meta| {
+            let q_scalar_fixed_short = meta.query_selector(self.q_scalar_fixed_short);
+            let sign = meta.query_advice(self.super_config.window, Rotation::cur());
+
+            vec![
+                q_scalar_fixed_short
+                    * (sign.clone() + Expression::Constant(C::Base::one()))
+                    * (sign - Expression::Constant(C::Base::one())),
```

Comment:
Simpler might be `q_scalar_fixed_short * (sign.clone() * sign.clone() - Expression::Constant(C::Base::one()))` (aka check that the sign bit squared equals one)

_Originally posted by @ebfull in https://github.com/zcash/orchard/pull/107#discussion_r647654106_